tet_mesh_scripts/coils_tet_mesh_script.py

```diff
@@ -0,0 +1,161 @@
+        row_extraction[0][0] = row_extraction[0][0].split('-\n')[1]
+
+        data_lists = [function_names, averages, std_devs, minima, maxima]
+        indices = np.arange(5)
+        for index, data_list in zip(indices, data_lists):
+            if index == 0:
+                data_list.append(row_extraction[index][0])
+            else:
+                data_list.append(float(row_extraction[index][0]))
+
+    # The above for-loop extracts the function names each as one word
+    # In order to be able to later match-case with Cubit's libary, the following for-loop separates function names to multiple words if applicable
+    split_list = []
+
+    for name in function_names:
+        split_words = []
+        current_word = ""
+
+        for char in name:
+            if char.isupper() and current_word.isupper() and len(current_word) > 1:
+                split_words.append(current_word)
+                current_word = char
+            elif char.isupper() and current_word and not current_word.isupper():
+                split_words.append(current_word)
+                current_word = char
+            else:
+                current_word += char
+
+        split_words.append(current_word)
+        split_list.append(" ".join(split_words))
+    function_names = split_list
+
+
+    # Compile the .log data into a Pandas DataFrame
+    quality_metrics_df = pd.DataFrame({'Function Name': function_names,
+                                    'Average': averages,
+                                    'Standard Deviation': std_devs,
+                                    'Minimum': minima,
+                                    'Maximum': maxima})
+
+    # Clean the dataframe of implausible entries (e.g. found a standard dev of 0 -- seemed too unlikely)
+    quality_metrics_df = quality_metrics_df.loc[quality_metrics_df['Standard Deviation'] != 0]
+
+    # Sort data to find the highest average mesh quality
+    quality_metrics_df = quality_metrics_df.sort_values('Average', ascending = False).reset_index()
+    best_analytic = quality_metrics_df['Function Name'][0]
+    print(best_analytic)
+
+
+    # MESH EXPORT
+
+    # Recreate the geometry and mesh with the best_analytic
+    cubit.cmd('reset')
+    cubit.cmd(f'import step "{coils_path}" heal')
+    cubit.cmd('volume  size auto factor 5')
+    cubit.cmd(f'volume {coil_num} scheme tetmesh')
+    cubit.cmd(f'volume {coil_num} scheme tetmesh')
+    cubit.cmd(f'mesh volume {coil_num}')
+    cubit.cmd(f'mesh volume {coil_num}')
+    cubit.cmd(f'quality volume {coil_num} {best_analytic} global draw mesh list')
+
+    # Export the mesh as a .exo file
+    exo_path = f"/home/eitan-weinstein/cubit/files_and_outputs/exo/coils/coils_tet_mesh_{coil_num}.exo"
+    cubit.cmd(f'export mesh "{exo_path}"')
+
+    # Convert to h5m
+    h5m_path = f"/home/eitan-weinstein/cubit/files_and_outputs/h5m/coils/coils_tet_mesh_{coil_num}.h5m"
+    h5m_conversion = sbp.run(f'mbconvert {exo_path} {h5m_path}', shell = True)
```

Comment:
Paths. You can consider including the final export path as an additional argument, though outputting the H5M file to the working directory is fine as well.